core/graph/src/main/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherAuthorisedGraphHandler.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.operation.export.graph.handler;
+
+import uk.gov.gchq.gaffer.graph.Graph;
+import uk.gov.gchq.gaffer.operation.export.graph.ExportToOtherAuthorisedGraph;
+import uk.gov.gchq.gaffer.operation.export.graph.OtherGraphExporter;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.library.GraphLibrary;
+import uk.gov.gchq.gaffer.store.operation.handler.export.ExportToHandler;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import uk.gov.gchq.gaffer.user.User;
+import uk.gov.gchq.koryphe.ValidationResult;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ExportToOtherAuthorisedGraphHandler extends ExportToHandler<ExportToOtherAuthorisedGraph, OtherGraphExporter> {
+
+    private Map<String, List<String>> idAuths = new HashMap<>();
+
+    public Map<String, List<String>> getIdAuths() {
+        return idAuths;
+    }
+
+    public void setIdAuths(final Map<String, List<String>> idAuths) {
+        if (idAuths == null) {
+            this.idAuths = new HashMap<>();
+        } else {
+            this.idAuths = idAuths;
+        }
+    }
+
+    @Override
+    protected Class<OtherGraphExporter> getExporterClass() {
+        return OtherGraphExporter.class;
+    }
+
+    @Override
+    protected OtherGraphExporter createExporter(final ExportToOtherAuthorisedGraph export, final Context context, final Store store) {
+        return new OtherGraphExporter(context.getUser(), createGraph(export, context, store));
+    }
+
+    protected Graph createGraph(final ExportToOtherAuthorisedGraph export, final Context context, final Store store) {
+        final String exportGraphId = export.getGraphId();
+        final String exportParentSchemaId = export.getParentSchemaId();
+        final String exportParentStorePropertiesId = export.getParentStorePropertiesId();
+        final GraphLibrary graphLibrary = store.getGraphLibrary();
+        final Graph graph;
+
+        validate(exportGraphId, exportParentSchemaId, exportParentStorePropertiesId, graphLibrary, store);
+
+        if (!isAuthorised(context.getUser(), idAuths.get(exportGraphId))) {
```

Comment:
We need to do all the authorisation first. If a user is not allow to export to graph1 then we should not attempt to load graph1 from the graphLibrary. This also applied to checking the user is authorised to use the schema ids and properties id.